lib/smart_proxy_reports/spooled_http_client.rb

```diff
@@ -52,37 +52,50 @@ def stop
       @worker.join
     end
 
+    def get_endpoint(name)
+      return "/api/v2/host_reports" if name[name.index("-") + 1] == "r"
+      return "/api/v2/hosts/facts"
+    end
+
+    def connect(http, factory)
+      processed = 0
+      files = Dir.glob(spool_path("todo", "*")).sort
+      files.each do |filename|
+        basename = File.basename(filename)
+        endpoint = get_endpoint(basename)
+        logger.debug "Sending report #{basename}"
+        begin
+          post = factory.create_post(endpoint, File.read(filename))
+          response = http.request(post)
+          logger.info "Report #{basename} sent with HTTP response #{response.code}"
+          logger.debug { "Response body: #{response.body}" }
+          if response.code.start_with?("2")
+            if Proxy::Reports::Plugin.settings.keep_reports
+              spool_move("todo", "done", basename)
+            else
+              FileUtils.rm_f spool_path("todo", basename)
+            end
+          else
+            logger.debug { "Moving failed report to 'fail' spool directory" }
+            spool_move("todo", "done", basename)
+          end
+          processed += 1
+        rescue StandardError => e
+          logger.warn "Unable to send #{basename}, will try on next request: #{e}", e
+          raise if ENV["RACK_ENV"] == "test"
+        end
+      end
+      processed
+    end
+
     def process
       processed = 0
       client = ::Proxy::HttpRequest::ForemanRequest.new
       factory = client.request_factory
       # send all files via a single persistent HTTP connection
       logger.debug "Opening HTTP connection to Foreman"
       client.http.start do |http|
-        Dir.glob(spool_path("todo", "*")) do |filename|
-          basename = File.basename(filename)
-          logger.debug "Sending report #{basename}"
-          begin
-            post = factory.create_post("/api/v2/host_reports", File.read(filename))
-            response = http.request(post)
-            logger.info "Report #{basename} sent with HTTP response #{response.code}"
-            logger.debug { "Response body: #{response.body}" }
-            if response.code.start_with?("2")
-              if Proxy::Reports::Plugin.settings.keep_reports
-                spool_move("todo", "done", basename)
-              else
-                FileUtils.rm_f spool_path("todo", basename)
-              end
-            else
-              logger.debug { "Moving failed report to 'fail' spool directory" }
-              spool_move("todo", "done", basename)
-            end
-            processed += 1
-          rescue StandardError => e
-            logger.warn "Unable to send #{basename}, will try on next request: #{e}", e
-            raise if ENV["RACK_ENV"] == "test"
-          end
-        end
+        processed += connect(http, factory)
```

Comment:
I know I told you to extract this into method, but can you put it back? Since we only call it once, its no longer necessary.